test/regression/external/submit-returns.spec.js

```diff
@@ -0,0 +1,440 @@
+  });
+});
+
+/**
+ * submit return by volmes measured
+ */
+describe('submit a return by volumes as an external user', function () {
+  it('sees the page title', async () => {
+    const viewLicencesLink = await $('*=View licences');
+    await viewLicencesLink.click();
+    const title = await getPageTitle();
+    expect(title).toHaveText('Your licences');
+  });
+
+  it('sees the licences table', async () => {
+    const table = await $('#results');
+    expect(table).toBeVisible();
+  });
+
+  it('sees the three licences created by the setup routine', async () => {
+    const table = await $('#results');
+    await expect(table).toHaveTextContaining('AT/CURR/MONTHLY/02');
+  });
+
+  it('clicks on the MONTHLY licence 02', async () => {
+    const dailyLicenceLink = await $('*=MONTHLY/02');
+    await dailyLicenceLink.click();
+    const licencePageHeader = await getPageTitle();
+    await expect(licencePageHeader).toBeDisplayed();
+    await expect(licencePageHeader).toHaveTextContaining('Licence number AT/CURR/MONTHLY/02');
+  });
+
+  it('sees the Summary table', async () => {
+    const table = await $('#summary');
+    expect(table).toBeVisible();
+  });
+
+  it('it clicks on the returns tab link', async () => {
+    const returnsTabLink = await $('*=Returns');
+    await returnsTabLink.click();
+  });
+
+  it('sees the returns table', async () => {
+    const table = await $('#returns');
+    expect(table).toBeVisible();
+  });
+
+  it('it clicks on the return id 9999991', () => {
+    const returnsLink = $('*=9999991');
+    returnsLink.click();
+    const pageTitle = getPageTitle();
+    expect(pageTitle).toHaveTextContaining('Abstraction return');
+    const caption = getPageCaption();
+    expect(caption).toHaveTextContaining('Licence number AT/CURR/MONTHLY/02');
+    expect($('form')).toBeVisible();
+  });
+
+  it('selects yes for water has been abstracted in this period', () => {
+    const radioButton1 = $('[value="false"]');
+    radioButton1.click();
+    const button = getButton();
+    button.click();
+  });
+
+  it('selects "Volumes from one or more meters"', () => {
+    expect($('form')).toBeVisible();
+    const radioButton2 = $('[value="abstractionVolumes,measured"]');
+    radioButton2.click();
+    const button = getButton();
+    button.click();
+  });
+
+  it('selects litres as the unit measured', () => {
+    expect($('form')).toBeVisible();
+    const formQuestion = $('//form/div/fieldset/legend');
+    expect(formQuestion).toHaveTextContaining('Which units are you using?');
+    const radioButton2 = $('[value="l"]');
+    radioButton2.click();
+    const button = getButton();
+    button.click();
+  });
+
+  it('enters the meter readings', () => {
+    expect($('form')).toBeVisible();
+    const formQuestion = $('//form/p');
+    expect(formQuestion).toHaveTextContaining('Your abstraction volumes');
+    const textBoxJan = $('input[name="2020-01-01_2020-01-31"]');
+    textBoxJan.setValue('10000');
```

Comment:
It be good if it tested the logic on this page a bit more, e.g what happens if:

* All fields left empty
* Some fields left empty
* Non-numeric values
* Zero values
* Negative values
* Volumes entered in each different unit available - are they calculated correctly

Reply:
Added tests for the above except for non-numerics because the UI does not accept non-numeric values